### PR TITLE
Find definition for locally declared entity

### LINKED ISSUE #625: Find definition for locally declared entity
Find definition for declared local entity .

Given this XML file:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE article [
  <!ENTITY mdash "&#x2014;">
]>
<root>
  &mdash;&mdash;
</root>
```

Ctrl+click on `&mdash;`should find definition of `mdash `

LocalEntityDefinitionDemo

### PR DESCRIPTION
**Pull request by angelozerr**
Find definition for locally declared entity

Fixes #625

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DTDElementDecl.java [hunk 1]
```diff
@@ -22,6 +26,8 @@
  */
 public class DTDElementDecl extends DTDDeclNode {
 
+	private static Predicate<Character> isValidChar = (c) -> Character.isJavaIdentifierPart(c) || c == '-';
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DTDElementDecl.java:29]: final

Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DTDElementDecl.java]: good catch, fixed

**Comment by angelozerr** [comment]
Here a demo with this XML:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE article [
  <!ENTITY mdash "&#x2014;">
]>
<root>
  &mdash;&mdash;
</root>
```

LocalEntityDefinitionDemo

**Comment by xorye** [comment]
It seems that
https://github.com/eclipse/lemminx/blob/6d2256fd3b9a880022ab4c9ec5644d80fc186e39/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DTDElementDecl.java#L91

should be changed to:
`* <!ELEMENT n|ote (to,from,heading,body)> will return null. `

**Pull request merged**